### ISSUE TITLE
HTTP Session Conflicts

### ISSUE THREAD

**Issue by Progentator** [isislab/CTFd#18]
Just started experiencing this issue.  

Two machines start out identical (download Kali amd64 image).  Machine A logs into an account, for some reason when Machine B goes to the web portal it is now logged into that account aswell.  What is even more odd is when Machine B logs out, it does not log out Machine A.

I've reproduced this issue three times from the latest build.

**Comment by ColdHeat** [comment]
Can you provide some kind of verification? I feel as I'm going to have a lot of trouble reproducing this.

**Comment by Progentator** [comment]
Nevermind.  I'm an idiot.  What happened was I accessed the page from my the image I distributed to set the home page.  My thought was since I never logged into the page, the cookie would not be set.  When in fact it was set but login does not trigger any change to the cookie, so after the first.  So when I logged out, it killed the session but the session got regenerated due to all the other machines still using the same cookie.

Or atleast that's what I think happened.  Either way, ensuring there is no cookie in the template fixed all issues.  Perhaps, something should happen to the cookies upon login to prevent this scenario from happening?

Well hopefully that makes sense, moral of the story is I'm an idiot and should of investigated more before opening a ticket.  Thanks for the quick response.

**Comment by bburky** [comment]
Probably unrelated: but I managed to trigger similar behavior by forgetting to clear SESSION_FILE_DIR (/tmp/flask_session) when I deleted the database. When I generated a new database, the old sessions continued to work even though all the accounts were destroyed and recreated. I couldn't figure out why some old sessions were staying valid until I realized I needed to clear SESSION_FILE_DIR.

I think this might be a separate issue though. Not sure if you consider this a bug. I can try to reproduce this and file a separate issue if you want.
